app/views/spree/admin/orders/_invoice_table.html.haml

```diff
@@ -10,19 +10,35 @@
       %th{:align => "right", :width => "15%"}
         %h4= t(:invoice_column_price)
   %tbody
-    - @order.line_items.sort_by{ |li| li.product.name }.each do |item|
-      %tr
-        %td
-          = render 'spree/shared/line_item_name', line_item: item
-          %br
-          %small
-            %em= raw(item.variant.product.supplier.name)
-        %td{:align => "right"}
-          = item.quantity
-        %td{:align => "right"}
-          = item.included_tax > 0 ? item.display_included_tax : ""
-        %td{:align => "right"}
-          = item.display_amount_with_adjustments
+    - if @order.distributor.preferred_invoice_order_by_supplier == true
```

Comment:
You are adding complexity here and duplicating code. I would assign `@line_items` in the controller. There you can define the sort depending on the setting and you need to change only one line in these templates.

Good alternatives are adding a helper or defining a method on the LineItem class, for example `sorted_for_invoice`. That may actually be the best solution because you may be able to do the sorting in the database and have this as scope.

Reply:
Alright, I'll go with this 👍. Thanks!